numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -1676,6 +1676,8 @@ PyArray_CLEARFLAGS(PyArrayObject *arr, int flags)
 #define PyDataType_ISOBJECT(obj) PyTypeNum_ISOBJECT(((PyArray_Descr*)(obj))->type_num)
 #define PyDataType_HASFIELDS(obj) (((PyArray_Descr *)(obj))->names != NULL)
 #define PyDataType_HASSUBARRAY(dtype) ((dtype)->subarray != NULL)
+#define PyDataType_ISUNSIZED(dtype) ((dtype)->elsize == 0)
```

Comment:
This is one of those cases where `(0 == (dtype)->elsize)` is a bit safer. Many years ago there was an attempt to subvert linux with a bit of trick code someone slipped into the public branch that replaced a check, something like `user == 0`, by `user = 0`.  Site shutdown and a months long hunt for the attacker's pathway ensued. That said, I don't think this is worth worrying about, it just reminds me when I see it ;)